CRM/Core/Smarty/plugins/function.help.php

```diff
@@ -67,6 +67,12 @@ function smarty_function_help($params, &$smarty) {
     $name = trim(strip_tags($params['title']));
   }
 
+  if (empty($name)) {
+    // sometimes the htxt id is field_name like the field, sometimes field-name instead so we fix it
+    $fieldID = str_replace('-', '_', str_replace('id-', '', $params['id']));
```

Comment:
I'm going to suggest doing the `id-` part as either a `preg_replace(/^id-/`, or checking with substr that it's only removing ones at the start of the string, otherwise if it says something like `paid-amount` or `covid-vaccine` it won't work.

Reply:
Yes, have done.